pkg/daemon/daemon.go

```diff
@@ -123,6 +124,9 @@ type Daemon struct {
 	loggerSupportsJournal bool
 
 	drainer *drain.Helper
+
+	// Ensures we only have a single validateOnDiskState running at a time.
+	diskValidatorLock sync.Mutex
```

Comment:
`dn.booting` only covers the bootstrap case. My thought behind the mutex was to ensure that the periodic check doesn't run in the middle of an update.

Reply:
I think this runs whenever MCD pod restarts or recreated, otherwise validateOnDiskState() would never get called today when we perform update/upgrade and we would have never got any bugs filed around it. @yuqi-zhang what do you think?

Reply:
> I think this runs whenever MCD pod restarts
So this part we are not sure of. I think we were able to test a scenario where a machineconfig triggered update does not reset dn.booting flag, so the rebooted run did not re-check the state. Also now that we have non-rebooting updates, it becomes a bit harder to just validate on `booting`. Zack and I were thinking of de-coupling validation and booting in general, and maybe even move more items out of the check (see https://issues.redhat.com/browse/MCO-86)

Reply:
We only call `validateOnDiskState()` whenever the MCD restarts (inside `checkStateOnFirstRun`). After that, we [set `dn.booting` to false and it never gets run again until the MCD restarts](https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/daemon.go#L445-L461). I remember discussing this with @yuqi-zhang and we were both surprised that it only runs there.

Reply:
This seems correct bc it's only set to true here: https://github.com/openshift/machine-config-operator/blob/7c1ac8b51423448397bda3f349f56f6e94261b64/pkg/daemon/daemon.go#L251